.github/workflows/preview.yml

```diff
@@ -0,0 +1,45 @@
+name: Preview on Vercel
+
+on:
+  # Run this workflow whenever a pull request is opened, synchronized, or reopened
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+        # Checkout the repository
+      - name: Check Out Code
+        uses: actions/checkout@v2
+
+        # Deploy to Vercel
+      - name: Deploy to Vercel
+        env:
+          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+        run: |
+          npm install -g vercel
+          vercel deploy --target preview
```

Comment:
The command `vercel deploy --target preview` means it will deploy a preview of the changes without affecting the production build.

The workflow is:

1. New `PR ` opens
2. Build and deploy a **preview** of the changes
3. PR is `Merged`
4. Build and deploy to the **production**

Reply:
Okay, so this has nothing to do with the creation or existence of the `preview branch` right?

Reply:
No, there won't be any activity within the GitHub repository. Just, the deployment to **Vercel's** server is solely trigerred by the `GitHub Actions workflow`.

Reply:
Okay thanks for the information :)